nixos/modules/config/swap.nix

```diff
@@ -124,6 +124,17 @@ let
         internal = true;
       };
 
+      discard = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Use swapon -d, and if randomEncryption, make the encrypted mapping
+          with --allow-discards.
+
+          If you use encrypted.enable for this swap device, you should enable
+          there encrypted.allowDiscards.
```

Comment:
this could be done automatically, right ?

Reply:
hm I'm not sure how. Any suggestion?